Improve edit security UI for auditors

Luke-Oldenburg opened the pull request:
## Summary of the problem
<!-- Why are these changes being made? What problem does it solve? Link any related issues to provide more details. -->
Buttons and form elements weren't disabled for auditors.


## Describe your changes
<!-- Explain your thought process to the solution and provide a quick summary of the changes. -->



<!-- If there are any visual changes, please attach images, videos, or gifs. -->

Luke-Oldenburg committed:
Improve edit security UI for auditors

Luke-Oldenburg committed:
Indent

Luke-Oldenburg committed:
Update app/views/users/edit_security.html.erb

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Luke-Oldenburg committed:
Update app/views/users/edit_security.html.erb

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Luke-Oldenburg committed:
Update app/views/users/edit_security.html.erb

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Luke-Oldenburg committed:
Remove extra whitespace

garyhtou commented on app/views/users/_user_session.html.erb:
```diff
@@ -13,11 +13,11 @@
       <% end %>
     <% elsif !session.expired? %>
-      <%= link_to logout_session_users_path(user: @user, id: session.id), data: { turbo_method: :delete, turbo_frame: "_top" }, class: "muted tooltipped tooltipped--w z5", 'aria-label': "Sign out of this session" do %>
+      <%= link_to logout_session_users_path(user: @user, id: session.id), data: { turbo_method: :delete, turbo_frame: "_top" }, class: "muted tooltipped tooltipped--w z5", 'aria-label': "Sign out of this session", disabled: do %>
```
```suggestion
      <%= link_to logout_session_users_path(user: @user, id: session.id), data: { turbo_method: :delete, turbo_frame: "_top" }, class: "muted tooltipped tooltipped--w z5", 'aria-label': "Sign out of this session", disabled: !policy(@user).logout_session? do %>
```

garyhtou commented on app/views/users/edit_security.html.erb:
```diff
@@ -1,183 +1,199 @@
+          <%= render "user_session", session:, disabled: %>
         <% elsif session.respond_to? :authorization_count %>
-          <%= render "oauth_authorization", authorization: session %>
+          <%= render "oauth_authorization", authorization: session, disabled: %>
```
```suggestion
          <%= render "oauth_authorization", authorization: session %>
```

It appears `oauth_authorization` is not using it at the moment. we'll add it in another PR.

garyhtou committed:
Discard changes to app/views/webauthn_credentials/_webauthn_credential.html.erb

garyhtou committed:
Use `UserPolicy#logout_session?` for disabling log out button

garyhtou committed:
Remove disabled

We don't have a policy method for this; and when this button is rendered, it is always useable

garyhtou committed:
Remove disabled

It appears oauth_authorization is not using it at the moment. we'll add it in another PR.

garyhtou committed:
Remove disabled

User session should have its own disabled logic.

It should be based on `UserSessionPolicy#update?` instead of the current `UserPolicy#update?`
